File: java/src/test/java/route/SegmentTest.java
package route;

import org.json.JSONObject;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class SegmentTest {

    private Driver driver;
    private JSONObject segment = new JSONObject("{'name': 'Segment static (JAVA)',\n" +
            "'description': 'From Driver',\n" +
            "'isTest': true,\n" +
            "'type': 'static',\n" +
            "'expiration': '2026-08-08T12:11:00Z'}");
    private JSONObject segmentUpdate = new JSONObject("{'name': 'Segment static (JAVA)',\n" +
            "'description': 'From Driver UPDATED',\n" +
            "'isTest': true,\n" +
            "'type': 'static',\n" +
            "'expiration': '2026-08-08T12:11:00Z'}");
    private String id = "15890";
    private String targetId = "000OGLYL";

    @Before
    public void setUp() {
        JSONObject config = new JSONObject();

        config.put("xkey", "my_xkey");
        config.put("url", "http://127.0.0.1:8888/");
        driver = new Driver(config);
    }

    @Test
    public void testCreate() {
        JSONObject response = this.driver.segment.create(this.segment);
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testUpdate() {
        JSONObject response = this.driver.segment.update(this.id, this.segmentUpdate);
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testGet() {
        JSONObject response = this.driver.segment.get(this.id);
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testList() {
        JSONObject response = this.driver.segment.list();
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testAddTarget() {
        JSONObject response = this.driver.segment.addToSegment(this.id, this.targetId);
        assertEquals(response.getInt("statusCode"), 204);
    }

    @Test
    public void testRemoveTarget() {
        JSONObject response = this.driver.segment.remove(this.id, this.targetId);
        assertEquals(response.getInt("statusCode"), 204);
    }

    @Test
    public void testDelete() {
        JSONObject response = this.driver.segment.delete(this.id);
        assertEquals(response.getInt("statusCode"), 204);
    }
}

File: csharp/DriverTest/SegmentTest.cs
using System.Collections.Generic;
using System.Net;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using NP6Api;
using NP6Api.Models;

namespace DriverTest
{
    [TestClass]
    public class SegmentTest
    {
        private static Driver _driver;
        private static SegmentModel _segment;

        [ClassInitialize()]
        public static void ActionTestInitialize(TestContext testContext)
        {
            var config = new Dictionary<string, string>();

            config["xKey"] = "YOUR XKEY";
            config["url"] = "http://127.0.0.1:8888/";
            _driver = new Driver(config);
            _segment = new SegmentModel();
            _segment.Name = "Test Segment";
            _segment.Description = "test";
            _segment.IsTest = true;
            _segment.Type = "static";
            _segment.Expiration = "2026-08-08T12:11:00Z";
        }

        [TestMethod]
        public void TestCreate()
        {
            var response = _driver.Segment.Create(_segment);

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestUpdate()
        {
            _segment.Description = "updated";
            var response = _driver.Segment.Update("12345", _segment);

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestGet()
        {
            var response = _driver.Segment.Get("12345");

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestList()
        {
            var response = _driver.Segment.List();

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestDelete()
        {
            var response = _driver.Segment.Delete("12345");

            Assert.AreEqual((HttpStatusCode)204, response.StatusCode);
        }

        [TestMethod]
        public void TestAddToSegment()
        {
            var response = _driver.Segment.AddToSegment("12345", "1234ABC");

            Assert.AreEqual((HttpStatusCode)204, response.StatusCode);
        }

        [TestMethod]
        public void TestRemove()
        {
            var response = _driver.Segment.Remove("12345", "1234ABC");

            Assert.AreEqual((HttpStatusCode)204, response.StatusCode);
        }
    }
}